src/lib/components/UrlChooser/index.js

```diff
@@ -0,0 +1,185 @@
+import {html} from "lit-element";
+import {store} from "../../store";
+import {BaseElement} from "../BaseElement";
+import {userRef, firebase} from "../../fb";
+
+/* eslint-disable require-jsdoc */
+class UrlChooser extends BaseElement {
+  static get properties() {
+    return {
+      href: {type: String},
+      switching: {type: Boolean, reflect: true},
+      active: {type: Boolean},
+    };
+  }
+
+  constructor() {
+    super();
+    this.href = null; // when signed out or waiting for Firestore, this is null
+    this.switching = true;
+    this.active = false;
+
+    // non-properties
+    this.urlInput = null;
+    this.runLighthouseButton = null;
+
+    store.subscribe(this.onStateChanged.bind(this));
+    this.onStateChanged();
+  }
+
+  render() {
+    return html`
+      <div class="report_header_enterurl">
+        <div class="lh-enterurl lh-enterurl--selected">${this.href}</div>
+        <div class="lh-enterurl lh-enterurl--switch">
+          <input
+            type="url"
+            class="lh-input"
+            .value="${this.href}"
+            placeholder="Enter a web page URL"
+            pattern="https?://.*"
+            minlength="7"
+            required
```

Comment:
Remove required per https://github.com/GoogleChrome/web.dev/issues/1180
I think(?) required is only supposed to be added after a validation check has failed.

Reply:
Done